core/chains/evm/gas/models.go

```diff
@@ -70,19 +70,53 @@ type DynamicFee struct {
 	TipCap *assets.Wei
 }
 
-type PriorAttempt interface {
-	GetGasPrice() *assets.Wei
-	DynamicFee() DynamicFee
+type PriorAttempt[FEE any, HASH any] interface {
```

Comment:
Are all the below functions for PriorAttempt needed by generic components?
I see most are only needed by the EVM gas estimators.

Could we try to expose here only the minimum set of fields?

Reply:
Sounds good. Let us look at it while doing the TxAttemtBuilder.